src/Cli/func/Helpers/AzureHelper.cs

```diff
@@ -383,31 +383,61 @@ internal static async Task CheckFunctionHostStatusForFlex(Site functionApp, stri
             functionAppReadyClient.DefaultRequestHeaders.Add("User-Agent", Constants.CliUserAgent);
             functionAppReadyClient.DefaultRequestHeaders.Add("Accept", jsonContentType);
 
+            var statusUri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+            ColoredConsole.WriteLine($"Target endpoint: {statusUri.GetLeftPart(UriPartial.Path)}");
+
+            int attemptCount = 1;
             await RetryHelper.Retry(
                 async () =>
                 {
+                    // Remove the header first if it already exists from a previous retry
+                    if (functionAppReadyClient.DefaultRequestHeaders.Contains("x-ms-request-id"))
+                    {
+                        functionAppReadyClient.DefaultRequestHeaders.Remove("x-ms-request-id");
+                    }
+
                     functionAppReadyClient.DefaultRequestHeaders.Add("x-ms-request-id", Guid.NewGuid().ToString());
-                    var uri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+
                     var request = new HttpRequestMessage()
                     {
-                        RequestUri = uri,
-                        Method = HttpMethod.Get
+                        RequestUri = statusUri,
+                        Method = HttpMethod.Get,
                     };
 
-                    var response = await functionAppReadyClient.SendAsync(request);
-                    ColoredConsole.Write(".");
+                    ColoredConsole.WriteLine($"Attempt {attemptCount}: Calling admin/host/status endpoint...");
```

Comment:
We log the retry # in the helper already so I don't think we need this line

`ColoredConsole.Error.WriteLine(ErrorColor($"Retry: {totalRetries - retryCount} of {totalRetries}"));`